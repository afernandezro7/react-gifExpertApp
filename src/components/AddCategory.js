import React, { useState } from 'react';
import PropTypes from 'prop-types';



export const AddCategory = ({setCategories}) => {

    const [inputValue, setinputValue] = useState('Introduce un Elemento')

    const handleInputChange= (e)=>{
        setinputValue(e.target.value);
    }

    const handleKeyEnter= (e)=>{
        e.preventDefault();

        if( inputValue.trim().length > 2){
            
            setCategories(cats=>[inputValue,...cats])
            setinputValue('Introduce un Elemento')
            document.getElementById('Myform').reset()
        }
    }

    return (
        
        <form id="Myform" onSubmit= { handleKeyEnter }>
            <input
                type="text"               
                placeholder={inputValue}
                
                onChange={ handleInputChange }
            />
        </form>
        
    )
}

AddCategory.propTypes= {
    setCategories: PropTypes.func.isRequired
}

