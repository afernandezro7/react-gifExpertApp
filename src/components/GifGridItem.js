import React from 'react'

export const GifGridItem = ({id,title,url}) => {
    return (    
        <div className="card animate__animated animate__rollIn animate__delay-2s">
            <img src={ url } alt={ title } />
            <p>{ title }</p>
           
        </div>  
    )
}
