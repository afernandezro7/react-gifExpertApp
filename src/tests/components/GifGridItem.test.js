import '@testing-library/jest-dom';
import React from 'react';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';


describe('Prueba sobre GifGridItem', () => {
    
    const title = 'un titulo';
    const url = 'https://unaurl.com';
    const wrapper = shallow(<GifGridItem title={title} url = {url} />)
    

    test('debe mostrar el componente correctamente ', () => {   
        expect( wrapper ).toMatchSnapshot()
    })

    test('Debe tener un parafo con el title', () => {
        const p = wrapper.find('p').text().trim()

        expect(p).toBe(title)
    })

    test('Debe tener la img con la url enviada y el alt', () => {
        const img = wrapper.find('img');
        //console.log(img.props());
        expect(img.props().src).toBe(url);
        expect(img.prop('alt')).toBe(title);
    })

    test('Debe de tener animate__rollIn', () => {
        const div = wrapper.find('div')
        
        expect(div.prop('className').includes('animate__rollIn')).toBe(true)
        //expect(div.html().includes('animate__rollIn')).toBe(true)

    })
    
    
    
    
})
