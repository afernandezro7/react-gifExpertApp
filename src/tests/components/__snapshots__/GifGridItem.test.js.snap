// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba sobre GifGridItem debe mostrar el componente correctamente  1`] = `
<div
  className="card animate__animated animate__rollIn animate__delay-2s"
>
  <img
    alt="un titulo"
    src="https://unaurl.com"
  />
  <p>
    un titulo
  </p>
</div>
`;
